### PR TITLE
SOLR-15703: replace all SolrException.log usage in Solr to just call log.error(...) directly

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-15703

### CODE REVIEW

File: solr/solrj/src/java/org/apache/solr/common/SolrException.java [hunk 1]
```diff
@@ -139,6 +140,7 @@ public String getRootThrowable() {
    * @deprecated Use the Logger directly
    */
   @Deprecated
+  @SuppressForbidden(reason = "self referencing")
   public void log(Logger log) {
```
Comment (risdenk) [inline solr/solrj/src/java/org/apache/solr/common/SolrException.java:144]: I'm going to remove all of these `log` methods in main once this is merged.

File: solr/core/src/java/org/apache/solr/core/SolrCore.java [hunk 2]
```diff
@@ -3013,10 +3013,6 @@ public static void postDecorateResponse(
     }
   }
 
-  public static final void log(Throwable e) {
-    SolrException.log(log, null, e);
-  }
```
Comment (risdenk) [inline solr/core/src/java/org/apache/solr/core/SolrCore.java:3018]: I can leave this to cleanup w/ the other SolrException.log, but this was only used in 3 places so just cleaned it up.

File: solr/core/src/java/org/apache/solr/core/ZkContainer.java [hunk 3]
```diff
@@ -219,10 +219,10 @@ public void registerInZk(final SolrCore core, boolean background, boolean skipRe
             } catch (InterruptedException e) {
               // Restore the interrupted status
               Thread.currentThread().interrupt();
-              SolrException.log(log, "", e);
+              log.error("", e);
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/core/ZkContainer.java:222]:
```suggestion
              log.error("Interrupted", e);
```

BTW I don't get why we had all these logs with empty strings.  On the consuming side as a user, it's weird to see when these exceptions happen.

File: solr/core/src/java/org/apache/solr/core/ZkContainer.java [hunk 4]
```diff
@@ -219,10 +219,10 @@ public void registerInZk(final SolrCore core, boolean background, boolean skipRe
...
             } catch (KeeperException e) {
-              SolrException.log(log, "", e);
-            } catch (AlreadyClosedException e) {
+              log.error("", e);
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/core/ZkContainer.java:224]:
```suggestion
              log.error("KeeperException", e);
```

File: solr/core/src/java/org/apache/solr/core/ZkContainer.java [hunk 5]
```diff
@@ -233,7 +233,7 @@ public void registerInZk(final SolrCore core, boolean background, boolean skipRe
               } catch (Exception e1) {
                 log.error("", e1);
               }
-              SolrException.log(log, "", e);
+              log.error("", e);
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/core/ZkContainer.java:236]:
```suggestion
              log.error(e.toString(), e);
```

**Review by dsmiley — COMMENTED** [review]
Thanks for doing this!  Nice to see thoughtful new error strings.

**Comment by risdenk** [comment]
There are a bunch of outstanding no error message logs as well:

```
# git grep -F 'log.error(""'
solr/core/src/java/org/apache/solr/cloud/Overseer.java:          log.error("", e);
solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java:        log.error("", e);
solr/core/src/java/org/apache/solr/cloud/RecoveryStrategy.java:        log.error("", e);
solr/core/src/java/org/apache/solr/cloud/ZkController.java:      log.error("", e);
solr/core/src/java/org/apache/solr/cloud/ZkController.java:      log.error("", e);
solr/core/src/java/org/apache/solr/cloud/ZkController.java:      log.error("", e);
solr/core/src/java/org/apache/solr/cloud/ZkController.java:      log.error("", e);
solr/core/src/java/org/apache/solr/cloud/ZkController.java:      log.error("", e);
solr/core/src/java/org/apache/solr/cloud/ZkController.java:      log.error("", e);
solr/core/src/java/org/apache/solr/cloud/ZkController.java:      log.error("", e);
solr/core/src/java/org/apache/solr/core/ZkContainer.java:        log.error("", e);
solr/core/src/java/org/apache/solr/core/ZkContainer.java:        log.error("", e);
solr/core/src/java/org/apache/solr/core/ZkContainer.java:              log.error("", e);
solr/core/src/java/org/apache/solr/core/ZkContainer.java:              log.error("", e);
solr/core/src/java/org/apache/solr/core/ZkContainer.java:                log.error("", e1);
solr/core/src/java/org/apache/solr/core/ZkContainer.java:                log.error("", e1);
solr/core/src/java/org/apache/solr/core/ZkContainer.java:              log.error("", e);
solr/core/src/java/org/apache/solr/filestore/DistribPackageStore.java:      log.error("", e);
solr/core/src/java/org/apache/solr/schema/ZkIndexSchemaReader.java:        log.error("", e);
solr/core/src/test/org/apache/solr/cloud/TestStressInPlaceUpdates.java:                log.error("", e);
solr/core/src/test/org/apache/solr/cloud/TestStressInPlaceUpdates.java:                log.error("", e);
solr/core/src/test/org/apache/solr/search/TestReloadDeadlock.java:                log.error("", e);
solr/core/src/test/org/apache/solr/search/TestStressReorder.java:                log.error("", e);
solr/core/src/test/org/apache/solr/search/TestStressReorder.java:                log.error("", e);
solr/core/src/test/org/apache/solr/search/TestStressUserVersions.java:                log.error("", e);
solr/core/src/test/org/apache/solr/search/TestStressUserVersions.java:                log.error("", e);
solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ConnectionManager.java:                    log.error("", e);
solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/SolrZkClient.java:        log.error("", e);
solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpClientUtil.java:                log.error("", e);
solr/solrj/src/java/org/apache/solr/common/util/ObjectReleaseTracker.java:          log.error("", t);
solr/solrj/src/java/org/apache/solr/common/util/ObjectReleaseTracker.java:          log.error("", t);
```

I also like @dsmiley don't understand why. I tried to change the SolrException.log("") to have some meaningful error message associated with it. I did not change any of the above.

**Pull request merged**